### PR TITLE
Bluetooth: Mesh: Make Sensor Observer fit nrf52832

### PR DESCRIPTION
**Pull request by ludvigsj**
This lowers the RAM requirements for the Sensor Observer sample so that it fits on the nRF52832. This is done by:

  * Reverting stack sizes and ADV buffer counts to default (tested and works fine).
  * Disabling the `FLASH_SHELL` module, which allocates 8k of buffers for a flash testing command the sample doesn't need.

New RAM usage after these changes: 61182 B (93.36% of nRF52832).

### CODE REVIEW

**Review by PavelVPV — COMMENTED** [review]
Have you checked the sample with our system test?

**Comment by ludvigsj** [comment]
> Have you checked the sample with our system test?

I have not. That test runs on the nRF52840 only, on CI. Should I manually run it on nRF52832 now, just to check?

**Comment by PavelVPV** [comment]
> > Have you checked the sample with our system test?
> 
> I have not. That test runs on the nRF52840 only, on CI. Should I manually run it on nRF52832 now, just to check?

It would make sense I think if it doesn't take much time. Does it pass on 52840 and 5340 after this change?

**Comment by ludvigsj** [comment]
> > > Have you checked the sample with our system test?
> > 
> > 
> > I have not. That test runs on the nRF52840 only, on CI. Should I manually run it on nRF52832 now, just to check?
> 
> It would make sense I think if it doesn't take much time. Does it pass on 52840 and 5340 after this change?

Passes on 52840. Is not marked as supported for nrf53.

**Comment by PavelVPV** [comment]
> > > > Have you checked the sample with our system test?
> > > 
> > > 
> > > I have not. That test runs on the nRF52840 only, on CI. Should I manually run it on nRF52832 now, just to check?
> > 
> > 
> > It would make sense I think if it doesn't take much time. Does it pass on 52840 and 5340 after this change?
> 
> Passes on 52840. Is not marked as supported for nrf53.

But where is the build?

**Comment by ludvigsj** [comment]
> > > > > Have you checked the sample with our system test?
> > > > 
> > > > 
> > > > I have not. That test runs on the nRF52840 only, on CI. Should I manually run it on nRF52832 now, just to check?
> > > 
> > > 
> > > It would make sense I think if it doesn't take much time. Does it pass on 52840 and 5340 after this change?
> > 
> > 
> > Passes on 52840. Is not marked as supported for nrf53.
> 
> But where is the build?

The test was run on the desk (which was easier than creating a test repo branch just to run a single test, I think), if that's what you mean. The sample is built by twister as well, compile only test.

**Review by PavelVPV — APPROVED** [review]

**Review by m-alperen-sener — APPROVED** [review]
looks good.

**Comment by ludvigsj** [comment]
DNM until @mia-ko has had a look at my changelog entry

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -169,7 +169,9 @@ Bluetooth samples
 Bluetooth mesh samples
 ----------------------
 
-|no_changes_yet_note|
+* :ref:`bluetooth_mesh_sensor_client` sample:
+
+  * Fixed an issue where the sample would not fit in RAM on the nrf52dk_nrf52832 board.
```
Comment (mia-ko) [inline doc/nrf/releases/release-notes-changelog.rst:174]:
```suggestion
  * Fixed an issue with the sample not fitting into RAM size on the nrf52dk_nrf52832 board.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by mia-ko — APPROVED** [review]

**Pull request merged**